src/accounts.rs

```diff
@@ -250,42 +513,74 @@ impl AccountsDB {
             .collect()
     }
 
-    pub fn increment_transaction_count(&mut self, tx_count: usize) {
-        self.transaction_count += tx_count as u64
+    pub fn increment_transaction_count(&self, tx_count: usize) {
+        let mut tx = self.transaction_count.write().unwrap();
+        *tx += tx_count as u64;
     }
 
     pub fn transaction_count(&self) -> u64 {
-        self.transaction_count
+        let tx = self.transaction_count.read().unwrap();
+        *tx
     }
     pub fn account_values_slow(&self) -> Vec<(Pubkey, solana_sdk::account::Account)> {
-        self.accounts.iter().map(|(x, y)| (*x, y.clone())).collect()
+        let mut vec: Vec<(Pubkey, Account)> = vec![];
+        let fork: Fork = 0;
+        let rindex = self.index_info.index.read().unwrap();
+        let stores = self.storage.read().unwrap();
+        rindex.iter().for_each(|(p, x)| {
+            if let Some((id, index)) = x.0.get(&fork) {
+                let account = stores[*id]
+                    .appendvec
+                    .read()
+                    .unwrap()
+                    .get_account(*index)
+                    .unwrap();
+                vec.push((*p, account));
+            }
+        });
+        vec
     }
-    fn merge(&mut self, other: Self) {
-        self.transaction_count += other.transaction_count;
-        self.accounts.extend(other.accounts)
+    fn merge(&self, other: &Self) {
```

Comment:
Using HashMap seems to be easier to help distinguish whether its a new account or just replacing an existing one which might otherwise require searching the entire vec each time.

Reply:
@sambley, for reads there is a search for the highest fork equal to or below. And the number of forks is at most 32

Reply:
Store should take a fork u64 as well